### PR TITLE
Process tomography using pyQuil's measure_observables

### PR DESCRIPTION
**Pull request by mpharrigan**
Refactor process tomography to be compatible with pyQuil's operator expectation functionality. This accompanies https://github.com/rigetti/pyquil/pull/770 (and depends on it).

This pull request is similar to #4, but this is for process tomography.

I've also cleaned up some of the docstrings and re-vamped the notebook

### CODE REVIEW

File: tests/test_process_tomography.py [hunk 1]
```diff
@@ -66,95 +75,82 @@ def test_proj_to_tni():
...
 
+    # Generate random unitary
+    u_rand = haar_rand_unitary(2 ** 1, rs=qc.qam.rs)
+    process = Program().defgate("RandUnitary", u_rand)
+    process.inst([("RandUnitary", q) for q in qubits])
 
-def test_single_qubit_z(qvm):
-    qvm.qam.random_seed = 1
-    process = Program(RZ(pi, 0))
+    # process = Program()
```
Comment (kylegulshen) [inline tests/test_process_tomography.py]: remove commented code

File: forest_benchmarking/tomography.py [hunk 2]
```diff
@@ -815,9 +859,9 @@ def estimate_variance(results: List[ExperimentResult],
                       n_resamples: int = 40,
                       project_to_physical: bool = False) -> Tuple[float, float]:
     """
-    Use a simple bootstrap-like method to return an errorbar on some 
+    Use a simple bootstrap-like method to return an errorbar on some
     functional of the quantum state.
-    
+
     :param exp_data: A namedtuple, called 'state_tomography_experiment_data'.
```
Comment (kylegulshen) [inline forest_benchmarking/tomography.py]: We need to update this from before.
exp_data -> results
+ qubits

File: tests/test_process_tomography.py [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
-from pyquil.gates import *
+from pyquil.api import QuantumComputer
+from pyquil.api._compiler import _extract_attribute_dictionary_from_program
+from pyquil.api._qac import AbstractCompiler
+from pyquil.device import NxDevice
+from pyquil.operator_estimation import measure_observables
+from pyquil.pyqvm import PyQVM
 
 REVERSE_CNOT_KRAUS = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]])
 CNOT_KRAUS = np.array([[1, 0, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0], [0, 1, 0, 0]])
```
Comment (kylegulshen) [inline tests/test_process_tomography.py]: We should remove these; they are no longer used, and the labels might confuse someone.

File: tests/test_process_tomography.py [hunk 4]
```diff
@@ -66,95 +75,82 @@ def test_proj_to_tni():
...
-    qvm.qam.random_seed = 1
-    process = Program(RX(pi / 2, 0)).inst(RZ(1, 0))
 
-    exp_desc = generate_process_tomography_experiment(process)
-    exp_data = acquire_tomography_data(exp_desc, qvm, var=.005)
-    estimate = pgdb_process_estimate(exp_data)
-    rx = expm(-1j * pi / 4 * sigma_x)
-    rz = expm(-1j / 2 * sigma_z)
-    assert np.allclose(estimate.estimate.process_choi_est, kraus2choi(rz @ rx), atol=.01)
+def test_single_q_linear_inversion(single_q_tomo_fixture):
```
Comment (kylegulshen) [inline tests/test_process_tomography.py]: This test failed both times locally.

**Comment by mpharrigan** [comment]
I'm going to nuke the linear inversion. It isn't working for me either ... was using it during testing though

**Pull request merged**